subsys/bluetooth/host/hci_core.c

```diff
@@ -853,41 +853,41 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 	}
 #endif
 
-	if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
-		if (evt->status) {
-			/*
-			 * if there was an error we are only interested in pending
-			 * connection so there is no need to check ID address as
-			 * only one connection can be in that state
-			 *
-			 * Depending on error code address might not be valid anyway.
-			 */
-			conn = find_pending_connect(NULL);
-			if (!conn) {
-				return;
-			}
-
-			conn->err = evt->status;
-
+	if (evt->status) {
 		/*
-		 * Handle advertising timeout after high duty directed
-		 * advertising.
+		 * If there was an error we are only interested in pending
+		 * connection. There is no need to check ID address as
+		 * only one connection can be in that state.
+		 *
+		 * Depending on error code address might not be valid anyway.
 		 */
-		if (conn->err == BT_HCI_ERR_ADV_TIMEOUT) {
-			atomic_clear_bit(bt_dev.flags, BT_DEV_ADVERTISING);
+		conn = find_pending_connect(NULL);
+		if (!conn) {
+			return;
 		}
 
-			bt_conn_set_state(conn, BT_CONN_DISCONNECTED);
+		conn->err = evt->status;
 
-			/* check if device is market for auto connect */
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
+			/*
+			 * Handle advertising timeout after high duty directed
+			 * advertising.
+			 */
+			if (conn->err == BT_HCI_ERR_ADV_TIMEOUT) {
+				atomic_clear_bit(bt_dev.flags, BT_DEV_ADVERTISING);
+			}
+		}
+
+		if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
```

Comment:
Prefer checking for relevant  HCI error code here.

Reply:
can you tell what you mean?